### PR TITLE
Update otelcontribcol image from 0.54.0 to 0.86.0 (#916)

### PR DESCRIPTION
**Pull request by victorpras**
The currently used version 0.54.0 has two fixable vulnerabilities, which requires to update jaeger to 1.47.0.
The openTelemetry-collector-contrib versions that are earlier than 0.85.0 use jaeger 1.41.0 or older.
Hence, this commit updates the image to the latest version, which unfortunately introduces breaking changes.

This commit includes the following changes:
- removed `--feature-gates=-exporter.googlecloud.OTLPDirect` because feature gate "exporter.googlecloud.OTLPDirect" is stable, can not be disabled.
- removed retry_on_failure because it was from the googlecloud exporter. The exporter itself handles retries, and retrying can cause issues. (#57233)
- added `--feature-gates=pkg.translator.prometheus.NormalizeName` to enable metric suffix trimming because the prometheus transformer appends `_ratio` to gauge metrics. link: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.86.0/pkg/translator/prometheus/normalize_name.go#L149
- updated the expected error message because otel-collector no longer has the message `failed to export time series to GCM`. See the old log[1] and the new log[2].

[1] old log:
```
2023-10-04T06:27:49.166Z	error	exporterhelper/queued_retry.go:149	Exporting failed. Try enabling retry_on_failure config option to retry on retryable errors	{"kind": "exporter", "data_type": "metrics", "name": "googlecloud/kubernetes", "error": "failed to export time series to GCM: rpc error: code = InvalidArgument desc = One or more TimeSeries could not be written: Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_name], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_name], [configsync_sync_namespace], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_namespace], [commit], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_kind], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_kind], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_name], [configsync_sync_namespace], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_name], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_name], [configsync_sync_namespace]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_name], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_kind], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_namespace], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_name], [configsync_sync_namespace], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_namespace], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_kind], [configsync_sync_name], [commit]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_name], [configsync_sync_namespace], [commit]; Unrecognized metric labels: [configsync_sync_name], [configsync_sync_namespace], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_name], [configsync_sync_kind], [configsync_sync_namespace]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_name], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_namespace], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_name], [commit], [configsync_sync_namespace], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_name], [configsync_sync_kind], [commit]; Unrecognized metric labels: [configsync_sync_name], [configsync_sync_namespace], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_name], [configsync_sync_namespace], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_namespace], [configsync_sync_name]; Unrecognized metric labels: [commit], [configsync_sync_name], [configsync_sync_namespace], [configsync_sync_kind]\nerror details: name = Unknown  desc = total_point_count:25 errors:{status:{code:3} point_count:25}"}
```

[2] new log:
```
2023-10-04T06:36:52.554Z	warn	batchprocessor@v0.86.0/batch_processor.go:258	Sender failed	{"kind": "processor", "name": "batch", "pipeline": "metrics/kubernetes", "error": "rpc error: code = InvalidArgument desc = One or more TimeSeries could not be written: Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_namespace], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_kind], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_name], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_name], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_name], [configsync_sync_namespace]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_namespace], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_kind], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_namespace], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_name], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_kind], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_namespace], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_namespace], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_name], [configsync_sync_namespace]; Unrecognized metric labels: [configsync_sync_kind], [configsync_sync_namespace], [commit], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_name], [configsync_sync_kind], [configsync_sync_namespace]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_name], [configsync_sync_kind]; Unrecognized metric labels: [configsync_sync_kind], [commit], [configsync_sync_namespace], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_kind], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_name], [configsync_sync_kind], [configsync_sync_namespace]; Unrecognized metric labels: [commit], [configsync_sync_name], [configsync_sync_kind], [configsync_sync_namespace]; Unrecognized metric labels: [configsync_sync_kind], [commit], [configsync_sync_namespace], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_name], [configsync_sync_kind], [configsync_sync_namespace]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_kind], [commit], [configsync_sync_name]; Unrecognized metric labels: [configsync_sync_name], [configsync_sync_kind], [commit], [configsync_sync_namespace]; Unrecognized metric labels: [configsync_sync_namespace], [configsync_sync_name], [configsync_sync_kind]\nerror details: name = Unknown  desc = total_point_count:25  errors:{status:{code:3}  point_count:25}"}
```

### CODE REVIEW

**Comment by victorpras** [comment]
The presubmit failures in `TestAutopilotReconcilerAdjustment` is caused by statically assigned validation values in the test, while in main branch the values are assigned dynamically. Some possible ways to fix this:
- Cherry pick https://github.com/GoogleContainerTools/kpt-config-sync/pull/876, https://github.com/GoogleContainerTools/kpt-config-sync/pull/910, and other PRs that are needed for the previous 2 PRs. This will changed the behavior of `TestAutopilotReconcilerAdjustment` to dynamically assigned validation values
- Fix the static values that is used in the validation. Changing `initialTotalCPU` and `initialTotalMemory` is all that's needed to get the test working.

**Comment by nan-yu** [comment]
I'm wondering why are the initial memory and CPU changed. The opentelemetry image update should not affect the resource, Do you know which container specifically?

**Comment by victorpras** [comment]
> I'm wondering why are the initial memory and CPU changed. The opentelemetry image update should not affect the resource, Do you know which container specifically?

I believe it's the hydration-controller container that was originally in the root-reconciler is now disabled by default

**Comment by nan-yu** [comment]
> > I'm wondering why are the initial memory and CPU changed. The opentelemetry image update should not affect the resource, Do you know which container specifically?
> 
> I believe it's the `hydration-controller`container that was originally in the root-reconciler is now disabled by default

How does it relate to the open telemetry change? Plus, that is not a new change. It behaves like that since 1.16.0.

**Comment by nan-yu** [comment]
I pulled your PR locally, and found out the resource diff is not coming from the hydration-controller, it is from the otel-agent container.
The resource requests and limits are deleted from your PR. It was not deleted in the original PR#916.

File: e2e/testcases/reconciler_manager_test.go [hunk 1]
```diff
@@ -52,9 +52,9 @@ import (
 const (
 	initialFirstCPU                    = 10
 	initialFirstMemory                 = 100
-	initialTotalCPU                    = 80
```
Comment (nan-yu) [inline e2e/testcases/reconciler_manager_test.go:55]: Please revert the change

File: manifests/templates/reconciler-manager-configmap.yaml [hunk 2]
```diff
@@ -181,18 +181,14 @@ data:
...
            command:
            - /otelcontribcol
            args:
            - "--config=/conf/otel-agent-config.yaml"
-           resources:
-             limits:
-               cpu: 1
-               memory: 1Gi
-             requests:
-               cpu: 10m
```
Comment (nan-yu) [inline manifests/templates/reconciler-manager-configmap.yaml:194]: Why is this removed? It is not removed in #916 that you're cherrypicking from.

**Comment by victorpras** [comment]
> How does it relate to the open telemetry change? Plus, that is not a new change. It behaves like that since 1.16.0.

I just checked out `v1.16.1` and observed the difference in test behavior between the 2 versions. In `v1.16.1`, the `root-reconciler` is originally created without `hydration-controller` container, but is soon evicted and recreated with `hydration-controller` container, so it passes the validation condition.

In this PR's presubmit, the `root-reconciler` is created without `hydration-controller` container and the pod is not recreated, so it never satisfies the validation. I'm not entirely sure yet on what's causing this different behavior with the new otel image

**Comment by nan-yu** [comment]
> I just checked out `v1.16.1` and observed the difference in test behavior between the 2 versions. In `v1.16.1`, the `root-reconciler` is originally created without `hydration-controller` container, but is soon evicted and recreated with `hydration-controller` container, so it passes the validation condition.

The observation is wrong. In both versions, the root reconciler is created without the hydration controller because that is the default behavior. The pod is evicted and recreated because in the test we check in DRY configs, which will enable the hydration controller.

File: manifests/templates/reconciler-manager-configmap.yaml (hunk 2 above)
Comment (victorpras) [reply manifests/templates/reconciler-manager-configmap.yaml]: looks like some merge conflict wasn't resolved correctly in the previous commit, should be fixed now

**Review by nan-yu — APPROVED** [review]
/lgtm
